scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,167 @@ def test_getc2_gesc2():
+    du = generate_random_dtype_array((n-1,), dtype=dtype)
+    A = np.diag(dl, -1) + np.diag(d) + np.diag(du, 1)
+    # generate random solution x
+    x = generate_random_dtype_array((n, 2), dtype=dtype)
+    # create b from x for equation Ax=b
+    if trans == 'N':
+        b = A @ x
+    else:
+        b = (A.conj().T) @ x
+
+    # store a copy of the inputs to check they haven't been modified later
+    inputs_cpy = [dl.copy(), d.copy(), du.copy(), b.copy()]
+
+    # determine dlf, df, duf, du2f, and IPIV from lambda expression
+
+    dlf_, df_, duf_, du2f_, ipiv_, info_lambda = dl_d_du_lambda(dl, d, du)
+
+    gtsvx_out = gtsvx(dl, d, du, b, fact=fact, trans=trans, dlf=dlf_, df=df_,
+                      duf=duf_, du2=du2f_, ipiv=ipiv_)
+    dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+    assert_(info == 0, "?gtsvx info = {}, should be zero".format(info))
+
+    # assure that inputs are unmodified
+    assert_array_equal(dl, inputs_cpy[0])
+    assert_array_equal(d, inputs_cpy[1])
+    assert_array_equal(du, inputs_cpy[2])
+    assert_array_equal(b, inputs_cpy[3])
+
+    # test that x_soln matches the expected x
+    assert_allclose(x, x_soln, rtol=rtol, atol=atol)
+
+    # assert that the outputs are of correct type or shape
+    # rcond should be a scalar
+
+    assert_(hasattr(rcond, "__len__") is not True,
+            "rcond should be scalar but is {}".format(rcond))
+    # ferr should be length of # of cols in x
+    assert_(ferr.shape[0] == b.shape[1], "ferr.shape is {} but shoud be {},"
+            .format(ferr.shape[0], b.shape[1]))
+    # berr should be length of # of cols in x
+    assert_(berr.shape[0] == b.shape[1], "berr.shape is {} but shoud be {},"
+            .format(berr.shape[0], b.shape[1]))
+
+    # test with singular matrix
+    # no need to test with fact "F" since ?gttrf already tests for these.
+    if fact == "N":
+
+        L, U = gttrf_to_lu(n, dtype, dlf, df, duf, du2f, ipiv)
+        # check that the outputs define an LU decomposition of A
+        assert_allclose(A, L @ U, atol=atol, rtol=rtol)
+        d[n-1] = 0
+        dl[n-2] = 0
+
+        # solve using routine
+        gtsvx_out = gtsvx(dl, d, du, b)
+        dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+        # test for the singular matrix.
+        assert_(d[info - 1] == 0,
+                "?gtsvx: d[info-1] is {}, not the illegal value"
+                .format(d[info - 1]))
+
+    elif fact == 'F':
+        # assuming that a singular factorization is input
+        df_[-1] = 0
+        duf_[n-2] = 0
+        du2f_[n-3] = 0
+
+        gtsvx_out = gtsvx(dl, d, du, b, fact=fact, dlf=dlf_, df=df_, duf=duf_,
+                          du2=du2f_, ipiv=ipiv_)
+        dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+        # info should not be zero and should provide index of illegal value
+        assert_(0 <= info < n, "incorrect info for singular matrix")
```

Comment:
```suggestion
        assert_(0 < info <= n, "incorrect info for singular matrix")
```
Right? We haven't subtracted 1 from `info` yet.